pkgs/development/go-modules/generic/default.nix

```diff
@@ -13,12 +13,12 @@
 # path to go.mod and go.sum directory
 , modRoot ? "./"
 
-# modSha256 is the sha256 of the vendored dependencies
+# vendorSha256 is the sha256 of the vendored dependencies
 #
-# CAUTION: if `null` is used as a value, the derivation won't be a
-# fixed-output derivation but disable the build sandbox instead. Don't use
-# this in nixpkgs as Hydra won't build those packages.
-, modSha256
+# if vendorSha256 is null, then we won't fetch any dependencies and # rely on the vendor folder within the source.
+, vendorSha256
+
+, modSha256 ? null
```

Comment:
This looks good in general. I would actually keep `modSha256` until after the next NixOS release. It adds a bit of bloat code temporally but helps users a lot to migrate their code. If `modSha256` is that a warning can be generated using `lib.warn` that instructs the user how to migrate to the new checksum or links to documentation. Speaking of which, these changes should be added to documentation as well as the releases notes.

Reply:
Where should the documentation be modified?

I only use nixpkgs unstable, so I really don't have a strong opinion about stable releases. It is going to annoying to have to keep this around for a year (2 releases rom now it'll get dropped), and I don't see why it really helps the user, they're going to have to make the migration either way, if they don't want to they can just stay on the old version. But I don't believe in or use stable so *shrug* added.

Reply:
The documentation is in `doc/languages-frameworks/go.xml`, it'd need to be updated to use `vendorSha256` instead of `modSha256`, and should explain what happens if it's unset.

As for release notes, a small explanation inside `sec-release-20.09-incompatibilities` of `nixos/doc/manual/release-notes/rl-2009.xml` explaining how this had changed, referring to the documentation should be enough.

People using stable and updating to the next release need to have some explanation when tooling changes.